pkg/services/ngalert/store/alert_rule.go

```diff
@@ -391,9 +415,33 @@ func (st DBstore) ValidateAlertRule(alertRule ngmodels.AlertRule, requireData bo
 func (st DBstore) UpdateRuleGroup(cmd UpdateRuleGroupCmd) error {
 	return st.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
 		ruleGroup := cmd.RuleGroupConfig.Name
+
+		// check if namespace exists
+		folderSrv := dashboards.NewFolderService(cmd.OrgID, cmd.RequestedBy, st.SQLStore)
+		folder, err := folderSrv.GetFolderBySlug(cmd.Namespace)
```

Comment:
Potential problem here if the namespace is renamed since that would create a new folder - what happens with the old folder and its rules?

Using `GetFolderBySlug` is not really recommended because the namespace will be matched against a slug optimized for representation in a URL. I wonder if this code works at all since the provided namespace is not "slugifed":
https://github.com/grafana/grafana/blob/33a4cfde510425989f72b3c55ac9f752d1c1f1ca/pkg/models/dashboards.go#L281
There's a chance multiple namespaces with special characters would result in the same slug. So really recommend using UID, if possible.

Reply:
> Potential problem here if the namespace is renamed since that would create a new folder - what happens with the old folder and its rules?

good catch! rules are associated with [the folder UID](https://github.com/grafana/grafana/blob/33a4cfde510425989f72b3c55ac9f752d1c1f1ca/pkg/services/ngalert/store/database_mig.go#L160) so in case of renaming the old folder and its rules will remain unmodified and a new folder with the old title and new new rules will be created (presumaby we will end up having a new folder with different (default) permissions and duplicate rules.

> Using GetFolderBySlug is not really recommended because the namespace will be matched against a slug optimized for representation in a URL. I wonder if this code works at all since the provided namespace is not "slugifed"

That's my wrong; I should have used title instead of slug which still is not the optimal (better always using UIDs)